api/nodes/views.py

```diff
@@ -1367,6 +1371,168 @@ def get_object(self):
         return fobj
 
 
+class NodeAddonList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin, NodeMixin):
+    """List of addons connected to this node *Read-only*
+
+    TODO: More Docs
+    """
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublic,
+        ExcludeRetractions,
+        base_permissions.TokenHasScope,
+    )
+
+    serializer_class = NodeAddonSettingsSerializer
+    view_category = 'nodes'
+    view_name = 'node-addons'
+
+    def get_default_queryset(self):
+        qs = []
+        for addon in ADDONS_OAUTH:
+            addon_settings = ADDONS_AVAILABLE_DICT[addon].settings_models.get('node', None)
+            obj = None
+            enabled = False
+            try:
+                obj = addon_settings.find(Q('owner', 'eq', self.get_node()._id) & Q('deleted', 'eq', False))[0]
+                enabled = True
+            except IndexError:
+                pass
+            qs.append(
+                {
+                    '_id': addon,
+                    'enabled': enabled,
+                    'settings': obj
+                }
+            )
+        return qs
+
+    get_queryset = get_default_queryset
+
+
+class NodeAddonDetail(JSONAPIBaseView, generics.RetrieveUpdateAPIView, NodeMixin):
+    """
+    TODO: docs
+    """
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublic,
+        ExcludeRetractions,
+        ReadOnlyIfRegistration,
+        base_permissions.TokenHasScope,
+    )
+
+    serializer_class = NodeAddonSettingsSerializer
+    view_category = 'nodes'
+    view_name = 'node-addon-detail'
+
+    def get_object(self):
+        addon = self.kwargs['provider']
+        if addon in ADDONS_OAUTH:
+            addon_settings = ADDONS_AVAILABLE_DICT[addon].settings_models.get('node', None)
+            obj = None
+            enabled = False
+            try:
+                obj = addon_settings.find(Q('owner', 'eq', self.get_node()) & Q('deleted', 'eq', False))[0]
+                enabled = True
+            except IndexError:
+                pass
+            return {
+                '_id': addon,
+                'enabled': enabled,
+                'settings': obj
+            }
+        raise NotFound('Requested addon unavailable.')
+
+class NodeAddonFolderList(JSONAPIBaseView, generics.ListAPIView, NodeMixin):
+    """
+    TODO: docs
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        ContributorOrPublic,
+        ExcludeRetractions,
+        base_permissions.TokenHasScope,
+    )
+
+    serializer_class = NodeAddonFolderSerializer
+    view_category = 'nodes'
+    view_name = 'node-addon-folders'
+
+    def _to_hgrid(self, item, addon, path):
+        """
+        :param item: contents returned from Google Drive API
```

Comment:
From [comment](https://github.com/CenterForOpenScience/osf.io/pull/5532#discussion_r61101918):
>As noted in this [`TODO`](https://github.com/CenterForOpenScience/osf.io/pull/5532/files#diff-51eef6ea2a6c35e2b080c9a6fe591e38R1487), the `<Addon>NodeSettings` models (and likely tests, generic views) require a moderate refactor in order to make [listing of available folders] generalizable via APIv2. Given that the only existing use case we have for this feature is for GDrive, and that it was requested a while ago, and that it will need post-model-refactoring refactoring, I wrote this part of the feature somewhat hastily.